### PR TITLE
Bi-monthly patching

### PR DESCRIPTION
**Pull request by hakoneriksson**
Need to apply CRDs for kube-prometheus-stack
https://artifacthub.io/packages/helm/prometheus-community/kube-prometheus-stack

Need to apply CRD for cert-manager
https://artifacthub.io/packages/helm/cert-manager/cert-manager/1.3.1

Need to apply CRDs for sealed-secrets
https://github.com/bitnami-labs/sealed-secrets/releases/tag/v0.16.0

Deleted clusterrole 
gitlab-prometheus-server (duplicates when trying to create helm release for gitlab)

We might also need to do more in monitoring, since this namespace was deleted in dev. Hopefully we only need to update it.

### CODE REVIEW

**Comment by hakoneriksson** [comment]
For discussion:
Should we upgrade AKS cluster from 1.9.7 at some point?

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ language: C
 env:
   global:
     - HELM_URL=https://get.helm.sh
-    - HELM_TAR=helm-v3.5.2-linux-amd64.tar.gz
+    - HELM_TAR=helm-v3.5.4-linux-amd64.tar.gz
```
Comment (audunsolemdal) [inline .travis.yml:7]: re-bump to 3.6.0, recent patch

File: base/gitlab/helm-release.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ spec:
   chart:
     repository: https://charts.gitlab.io/
     name: gitlab
-    version: 4.9.2
+    version: 4.12.0
```
Comment (audunsolemdal) [inline base/gitlab/helm-release.yaml:13]: Note that this is the only bump of any real risk of downtime. Consider doing this in a separate PR unless you are very certain of minimal downtime.

Comment (audunsolemdal) [reply base/gitlab/helm-release.yaml:13]: Otherwise, rebump to 4.12.1

File: base/standalone-components/external-dns/helm-release.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
   chart:
     repository: https://charts.bitnami.com/bitnami
     name: external-dns
-    version: 4.8.6
+    version: 5.0.0
```
Comment (audunsolemdal) [inline base/standalone-components/external-dns/helm-release.yaml:16]: rebump to 5.0.2

File: base/monitoring/prometheus-operator/helm-release.yaml [hunk 4]
```diff
@@ -24,6 +24,7 @@ spec:
           nginx.ingress.kubernetes.io/auth-signin: "https://$host/oauth2/start?rd=$escaped_request_uri"
         hosts:
         - alertmanager.sdpaks.equinor.com
+        pathType: ImplementationSpecific
```
Comment (audunsolemdal) [inline base/monitoring/prometheus-operator/helm-release.yaml:27]: explain the "pathType:" changes. What do they do?

**Comment by audunsolemdal** [comment]
> For discussion:
> Should we upgrade AKS cluster from 1.9.7 at some point?

From 1.19.7 yes.
In dev cluster you can upgrade to latest available preview version
In prod cluster - you can upgrade to latest non-preview version

Make sure before upgrading prod that you test the same exact version upgrade in dev.

**Comment by audunsolemdal** [comment]
In general - downtime is acceptable for certain charts, exceptions:

external-dns
gitlab
(partially): cert-manager
(partially): velero

Especially the top two you need to be comfortable of minimal downtime before merging.

**Comment by audunsolemdal** [comment]
velero is casting errors
some websites in dev without certificates

File: base/monitoring/prometheus-operator/helm-release.yaml (hunk 4 above)
Comment (hakoneriksson) [reply base/monitoring/prometheus-operator/helm-release.yaml:27]: https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types

ImplementationSpecific is the same as leaving it empty before. Only difference is that it is now required to have this field in the yaml.

File: base/gitlab/helm-release.yaml (hunk 2 above)
Comment (hakoneriksson) [reply base/gitlab/helm-release.yaml:13]: Agreed. Will do this in a separate PR.

File: .travis.yml (hunk 1 above)
Comment (hakoneriksson) [reply .travis.yml]: bumped.

**Review by audunsolemdal — APPROVED** [review]
Remember to apply crds for cert-manager before merging, otherwise we risk rate limiting of certs

**Comment by hakoneriksson** [comment]
Have applied CRDs for cert-manager and sealed-secrets. closing PR

**Pull request merged**